src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/impl/GitClientImpl.java

```diff
@@ -259,6 +260,62 @@ private boolean setConfig() {
 		return getDiff(revCommit, revCommit);
 	}
 
+	/**
+	 *
+	 * @param featureBranch
+	 *            ref of the feature branch
+	 */
+	@Override
+	public List<RevCommit> getFeatureBranchCommits(Ref featureBranch) {
+		List<RevCommit> branchUniqueCommits = new ArrayList<RevCommit>();
+		List<RevCommit> branchCommits = getCommits(featureBranch);
+		RevCommit lastCommonAncestor = null;
+		for (RevCommit commit : branchCommits) {
+			if (defaultBranchCommits.contains(commit)) {
+				LOGGER.info("Found last common commit "+commit.toString());
+				lastCommonAncestor = commit;
+				break;
+			}
+			branchUniqueCommits.add(commit);
+		}
+
+		if (lastCommonAncestor==null) {
+			branchUniqueCommits = null;
+		}
+		else {
+			branchUniqueCommits=Lists.reverse(branchUniqueCommits);
+		}
+
+		return branchUniqueCommits;
+	}
+
+	@Override
+	public List<RevCommit> getFeatureBranchCommits(String featureBranchName) {
+		Ref featureBranch = getBranch(featureBranchName);
+		if (null == featureBranch) {
+			return (List<RevCommit>)null;
```

Comment:
I would return an empty list in this case instead of null, but it might be a better practice to return null instead. @lw2011 and @dekome do you think it's better to return an empty list or null?

[issue] What is the return value of methods that would normally return a collection (e.g. list) with an invalid input parameter? [/issue]
[alternative] Methods with an invalid input parameter return an empty list! [/alternative]
[pro] Prevents a null pointer exception. [/pro]
[con] Is misleading since it is not clear whether the list is empty but has a valid input parameter or because of an invalid paramater. [/con]
[alternative] Methods with an invalid input parameter return null! [/alternative]